contracts/lib/Authorizable.sol

```diff
@@ -1,72 +1,51 @@
-/*
-
-  Copyright 2017 ZeroEx Intl.
-
-  Licensed under the Apache License, Version 2.0 (the "License");
-  you may not use this file except in compliance with the License.
-  You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-  Unless required by applicable law or agreed to in writing, software
-  distributed under the License is distributed on an "AS IS" BASIS,
-  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-  See the License for the specific language governing permissions and
-  limitations under the License.
-
-*/
-
 pragma solidity ^0.5.8;
 
-import "./Ownable.sol";
 
+contract Authorizable {
 
-contract Authorizable is Ownable {
-
-    /// @dev Only authorized addresses can invoke functions with this modifier.
-    modifier onlyAuthorized {
-        require(authorized[msg.sender]);
-        _;
-    }
-
-    modifier targetAuthorized(address target) {
-        require(authorized[target]);
-        _;
-    }
-
-    modifier targetNotAuthorized(address target) {
-        require(!authorized[target]);
-        _;
-    }
+    mapping (address => bool) public globalAuthorized;
+    // mapping from user to mapping Address => isAuthorized
+    mapping (address => mapping (address =>bool)) public authorized;
 
-    mapping (address => bool) public authorized;
+    event GlobalAuthorizedAddressAdd(address indexed target);
+    event GlobalAuthorizedAddressRemove(address indexed target);
 
-    event LogAuthorizedAddressAdded(address indexed target, address indexed caller);
-    event LogAuthorizedAddressRemoved(address indexed target, address indexed caller);
-
-    /*
-     * Public functions
-     */
+    event AuthorizedAddressAdd(address indexed target, address indexed sender);
+    event AuthorizedAddressRemove(address indexed target, address indexed sender);
 
     /// @dev Authorizes an address.
     /// @param target Address to authorize.
     function addAuthorizedAddress(address target)
         public
-        onlyOwner
-        targetNotAuthorized(target)
     {
-        authorized[target] = true;
-        emit LogAuthorizedAddressAdded(target, msg.sender);
+        authorized[msg.sender][target] = true;
+        emit AuthorizedAddressAdd(target, msg.sender);
     }
 
     /// @dev Removes authorizion of an address.
     /// @param target Address to remove authorization from.
     function removeAuthorizedAddress(address target)
         public
-        onlyOwner
-        targetAuthorized(target)
     {
-        delete authorized[target];
-        emit LogAuthorizedAddressRemoved(target, msg.sender);
+        delete authorized[msg.sender][target];
```

Comment:
Should we `require(authorized[msg.sender][target]);`?

Reply:
Yes, good catch. Same for global